src/GenSymIO.chpl

```diff
@@ -1908,16 +1929,16 @@ module GenSymIO {
                 firstSeg = 0;
                 lastSeg = segsArray.front();
             } else {                                                         
-                var (nullString,fSeg) = bytesArray.find(NULL_STRINGS_VALUE);
+                var (nullString,fSeg) = charArray.find(NULL_STRINGS_VALUE);
                 if nullString {
                     firstSeg = fSeg + 1;
                 }
                 lastSeg = segsArray.back();
             }
 
             /*
-             * Normalize the first and last seg elements (make then zero-based) by
-             * subtracting the bytes domain first element. 
+             * Normalize the first and last seg elements (make them zero-based) by
+             * subtracting the char domain first index element. 
 '            */
```

Comment:
Note: There is a dangling single quote here on line 1942 (it's in a comment so it doesn't really affect anything other than IDE code plugins).  Since there is some minor cleanup going on it's probably worth it to get that one to.